prompt_test.go

```diff
@@ -135,3 +135,19 @@ func TestDisableReverseSearch(t *testing.T) {
 	assert.Equal(t, "", prompt.buf.Text())
 	assert.Equal(t, len(history.tmp)-1, history.selected)
 }
+
+func TestPushToHistory(t *testing.T) {
+	prompt := &Prompt{
+		reverseSearchOpt: true,
+		history:          NewHistory(),
+		buf:              NewBuffer(),
+	}
+	prompt.disableAutoHistory = true
+	err := prompt.PushToHistory("cmd")
+	assert.NoError(t, err)
+	assert.Equal(t, len(prompt.history.histories), 1)
+
+	prompt.disableAutoHistory = false
+	err = prompt.PushToHistory("cmd")
+	assert.Error(t, err)
+}
```

Comment:
This is not a very good test, it is better to test the public API instead of the implementation.

Reply:
Fixed.